### ISSUE TITLE
ModuleNotFoundError: No module named 'hdf5plugin._config'

### ISSUE THREAD

**Issue by junyuewang** [silx-kit/hdf5plugin#122]
My machine architechture is mips64el, system is Deepin based on Debian 10, I successfully built it by disabling sse2 and avx2 and changing flag -mcup to -march.
```
sudo python3 setup.py install
running install
creating /tmp/tmpb4wuc4uy/tmp
creating /tmp/tmpb4wuc4uy/tmp/tmpb4wuc4uy
mips64el-linux-gnuabi64-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -c /tmp/tmpb4wuc4uy/source.cc -o /tmp/tmpb4wuc4uy/tmp/tmpb4wuc4uy/source.o -std=c++11
WARNING:__main__:SSE2 disabled: not available
WARNING:__main__:AVX2 disabled: not available
creating /tmp/tmp4p39ngq1/tmp
creating /tmp/tmp4p39ngq1/tmp/tmp4p39ngq1
mips64el-linux-gnuabi64-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -c /tmp/tmp4p39ngq1/source.c -o /tmp/tmp4p39ngq1/tmp/tmp4p39ngq1/source.o -fopenmp
setup.py:88: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
  "CPU info detection does not support this architecture: SSE2 and AVX2 disabled")
WARNING:__main__:CPU info detection does not support this architecture: SSE2 and AVX2 disabled
INFO:__main__:Building with C++11: True
INFO:__main__:Building with native option: True
INFO:__main__:Building with SSE2: False
INFO:__main__:Building with AVX2: False
INFO:__main__:Building with OpenMP: True
running bdist_egg
running egg_info
writing src/hdf5plugin.egg-info/PKG-INFO
writing dependency_links to src/hdf5plugin.egg-info/dependency_links.txt
writing requirements to src/hdf5plugin.egg-info/requires.txt
writing top-level names to src/hdf5plugin.egg-info/top_level.txt
INFO:root:Generating grammar tables from /usr/lib/python3.7/lib2to3/Grammar.txt
INFO:root:Generating grammar tables from /usr/lib/python3.7/lib2to3/PatternGrammar.txt
reading manifest file 'src/hdf5plugin.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '*__pycache__/*' found anywhere in distribution
writing manifest file 'src/hdf5plugin.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-mips64/egg
running build_clib
building 'snappy' library
building 'charls' library
building 'zfp' library
running install_lib
running build_py
running build_ext
creating build/bdist.linux-mips64/egg
creating build/bdist.linux-mips64/egg/hdf5plugin
creating build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/plugins/libh5bshuf.so -> build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/plugins/libh5blosc.so -> build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/plugins/libh5zfp.so -> build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/plugins/libh5fcidecomp.so -> build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/plugins/libh5lz4.so -> build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/plugins/libh5zstd.so -> build/bdist.linux-mips64/egg/hdf5plugin/plugins
copying build/lib.linux-mips64-3.7/hdf5plugin/__init__.py -> build/bdist.linux-mips64/egg/hdf5plugin
copying build/lib.linux-mips64-3.7/hdf5plugin/_version.py -> build/bdist.linux-mips64/egg/hdf5plugin
copying build/lib.linux-mips64-3.7/hdf5plugin/test.py -> build/bdist.linux-mips64/egg/hdf5plugin
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/_version.py to _version.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/test.py to test.cpython-37.pyc
creating stub loader for hdf5plugin/plugins/libh5lz4.so
creating stub loader for hdf5plugin/plugins/libh5bshuf.so
creating stub loader for hdf5plugin/plugins/libh5blosc.so
creating stub loader for hdf5plugin/plugins/libh5fcidecomp.so
creating stub loader for hdf5plugin/plugins/libh5zfp.so
creating stub loader for hdf5plugin/plugins/libh5zstd.so
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/plugins/libh5lz4.py to libh5lz4.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/plugins/libh5bshuf.py to libh5bshuf.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/plugins/libh5blosc.py to libh5blosc.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/plugins/libh5fcidecomp.py to libh5fcidecomp.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/plugins/libh5zfp.py to libh5zfp.cpython-37.pyc
byte-compiling build/bdist.linux-mips64/egg/hdf5plugin/plugins/libh5zstd.py to libh5zstd.cpython-37.pyc
creating build/bdist.linux-mips64/egg/EGG-INFO
copying src/hdf5plugin.egg-info/PKG-INFO -> build/bdist.linux-mips64/egg/EGG-INFO
copying src/hdf5plugin.egg-info/SOURCES.txt -> build/bdist.linux-mips64/egg/EGG-INFO
copying src/hdf5plugin.egg-info/dependency_links.txt -> build/bdist.linux-mips64/egg/EGG-INFO
copying src/hdf5plugin.egg-info/requires.txt -> build/bdist.linux-mips64/egg/EGG-INFO
copying src/hdf5plugin.egg-info/top_level.txt -> build/bdist.linux-mips64/egg/EGG-INFO
writing build/bdist.linux-mips64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
hdf5plugin.__pycache__.__init__.cpython-37: module references __file__
hdf5plugin.plugins.__pycache__.libh5blosc.cpython-37: module references __file__
hdf5plugin.plugins.__pycache__.libh5bshuf.cpython-37: module references __file__
hdf5plugin.plugins.__pycache__.libh5fcidecomp.cpython-37: module references __file__
hdf5plugin.plugins.__pycache__.libh5lz4.cpython-37: module references __file__
hdf5plugin.plugins.__pycache__.libh5zfp.cpython-37: module references __file__
hdf5plugin.plugins.__pycache__.libh5zstd.cpython-37: module references __file__
creating 'dist/hdf5plugin-3.0.0-py3.7-linux-mips64.egg' and adding 'build/bdist.linux-mips64/egg' to it
removing 'build/bdist.linux-mips64/egg' (and everything under it)
Processing hdf5plugin-3.0.0-py3.7-linux-mips64.egg
removing '/usr/local/lib/python3.7/dist-packages/hdf5plugin-3.0.0-py3.7-linux-mips64.egg' (and everything under it)
creating /usr/local/lib/python3.7/dist-packages/hdf5plugin-3.0.0-py3.7-linux-mips64.egg
Extracting hdf5plugin-3.0.0-py3.7-linux-mips64.egg to /usr/local/lib/python3.7/dist-packages
hdf5plugin 3.0.0 is already the active version in easy-install.pth

Installed /usr/local/lib/python3.7/dist-packages/hdf5plugin-3.0.0-py3.7-linux-mips64.egg
Processing dependencies for hdf5plugin==3.0.0
Searching for cached-property
Reading https://pypi.org/simple/cached-property/
Downloading https://files.pythonhosted.org/packages/48/19/f2090f7dad41e225c7f2326e4cfe6fff49e57dedb5b53636c9551f86b069/cached_property-1.5.2-py2.py3-none-any.whl#sha256=df4f613cf7ad9a588cc381aaf4a512d26265ecebd5eb9e1ba12f1319eb85a6a0
Best match: cached-property 1.5.2
Processing cached_property-1.5.2-py2.py3-none-any.whl
Installing cached_property-1.5.2-py2.py3-none-any.whl to /usr/local/lib/python3.7/dist-packages
Adding cached-property 1.5.2 to easy-install.pth file

Installed /usr/local/lib/python3.7/dist-packages/cached_property-1.5.2-py3.7.egg
Searching for h5py==3.2.1
Best match: h5py 3.2.1
Processing h5py-3.2.1-py3.7-linux-mips64.egg
h5py 3.2.1 is already the active version in easy-install.pth

Using /usr/local/lib/python3.7/dist-packages/h5py-3.2.1-py3.7-linux-mips64.egg
Searching for numpy==1.16.2
Best match: numpy 1.16.2
Adding numpy 1.16.2 to easy-install.pth file
Installing f2py script to /usr/local/bin
Installing f2py3 script to /usr/local/bin
Installing f2py3.7 script to /usr/local/bin

Using /usr/lib/python3/dist-packages
Finished processing dependencies for hdf5plugin==3.0.0
```
But when import it, got module not found error:
```
python3
Python 3.7.3 (default, Feb 26 2021, 06:12:32) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import hdf5plugin
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/hdf5plugin-3.0.0-py3.7-linux-mips64.egg/hdf5plugin/__init__.py", line 50, in <module>
    from ._config import config
ModuleNotFoundError: No module named 'hdf5plugin._config'
```
Could you help? Thank you very much!

**Comment by vasole** [comment]
Is the problem still there if you use pip3 instead of python3 setup.py to build and install the package?

**Comment by junyuewang** [comment]
now if I run pip3, it says requirement already satisfied
```
pip3 install hdf5plugin --user
Requirement already satisfied: hdf5plugin in /usr/local/lib/python3.7/dist-packages/hdf5plugin-3.0.0-py3.7-linux-mips64.egg (3.0.0)
Requirement already satisfied: h5py in /usr/local/lib/python3.7/dist-packages/h5py-3.2.1-py3.7-linux-mips64.egg (from hdf5plugin) (3.2.1)
Requirement already satisfied: cached-property in /usr/local/lib/python3.7/dist-packages/cached_property-1.5.2-py3.7.egg (from h5py->hdf5plugin) (1.5.2)
Requirement already satisfied: numpy>=1.14.5 in /usr/lib/python3/dist-packages (from h5py->hdf5plugin) (1.16.2)
```
but still produces the same error:
```
python3
Python 3.7.3 (default, Feb 26 2021, 06:12:32) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import hdf5plugin
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/hdf5plugin-3.0.0-py3.7-linux-mips64.egg/hdf5plugin/__init__.py", line 50, in <module>
    from ._config import config
ModuleNotFoundError: No module named 'hdf5plugin._config'
```
Before I compiled and built from source code, pip3 install would not go through as sse2, avx2 were not supported by my mips64el machine. So I manually changed setup.py to disable the feature.
Thanks!

**Comment by t20100** [comment]
Hi,

Do you build from the github repository or from the source archive from pypi.org?

**Comment by junyuewang** [comment]
I built it from github repository by git clone command. Thanks!

**Comment by t20100** [comment]
OK, issue is only when installing with `python setup.py install` without running `python setup.py build` before, in this case `Build.run()` is apparently not called (I didn't expected that!).

We'll find a fix, thanks for the report!

BTW, if you want to propose a PR to add support for `mips64el` that would be great!

From what you said, it should boils down to adding support for `mips64el` at those 2 places:
- https://github.com/silx-kit/hdf5plugin/blob/main/setup.py#L78
- https://github.com/silx-kit/hdf5plugin/blob/main/setup.py#L317

**Comment by junyuewang** [comment]
maybe the line 78 should be:
```
    if platform.machine() in ["arm64", "mips64"]:
        return False, False
```
and line 317:
```
                if platform.machine() in ["i386", "i686", "amd64", "x86_64", "mips64"]:
                    e.extra_compile_args += ['-march=native']
```
on my mips64el machine, platform.machine() returns mips64.
thank you all for the support!

**Comment by t20100** [comment]
Yes, that sounds good!
Do you mind testing it (from the source directory, after installing `hdf5plugin` (`pip uninstall hdf5plugin -y; pip install .`), run: `python test/test.py` and `python -c "import hdf5plugin.test; hdf5plugin.test.run_tests()"`) and proposing a Pull Request?

**Comment by junyuewang** [comment]
here is the output:
```
python3 test/test.py
testBitshuffle (__main__.TestHDF5PluginRead)
Test reading bitshuffle compressed data ... ok
testBlosc (__main__.TestHDF5PluginRead)
Test reading Blosc compressed data ... ok
testFicdecomp (__main__.TestHDF5PluginRead)
Test reading FICDECOMP compressed data ... ok
testLZ4 (__main__.TestHDF5PluginRead)
Test reading lz4 compressed data ... ok
testZfp (__main__.TestHDF5PluginRead)
Test reading ZFP compressed data ... ok
testBitshuffle (hdf5plugin.test.TestHDF5PluginRW)
Write/read test with bitshuffle filter plugin ... ok
testBlosc (hdf5plugin.test.TestHDF5PluginRW)
Write/read test with blosc filter plugin ... ok
testFciDecomp (hdf5plugin.test.TestHDF5PluginRW)
Write/read test with fcidecomp filter plugin ... ok
testLZ4 (hdf5plugin.test.TestHDF5PluginRW)
Write/read test with lz4 filter plugin ... ok
testZfp (hdf5plugin.test.TestHDF5PluginRW)
Write/read test with zfp filter plugin ... ok
testZstd (hdf5plugin.test.TestHDF5PluginRW)
Write/read test with Zstd filter plugin ... ok
testConfig (hdf5plugin.test.TestPackage)
Test hdf5plugin.config availability ... ok
testVersion (hdf5plugin.test.TestPackage)
Test version information ... ok

----------------------------------------------------------------------
Ran 13 tests in 1.730s

OK
```
and
```
python3 -c "import hdf5plugin.test; hdf5plugin.test.run_tests()"
........
----------------------------------------------------------------------
Ran 8 tests in 0.748s

OK
Test suite succeeded
```
thanks!

**Comment by t20100** [comment]
All good!

**Comment by t20100** [comment]
closed by #125 and #126
